src/teacher/templates/teacher/includes/pagination.html

```diff
@@ -0,0 +1,19 @@
+<div class="pagination" style="padding: 2rem 0">
+    <span class="step-links">
+
+        {% if page_obj.has_previous %}
+            <a href="?page=1" class="page-link" style="display: inline-block">&laquo; first</a>
+            <a href="?page={{ page_obj.previous_page_number }}" class="page-link" style="display: inline-block">previous</a>
+        {% endif %}
+
+        {% for number_page in page_obj.paginator.page_range %}
```

Comment:
In case of big number of elements and small page size the generated list of page links could be big indeed.
You can limit printed number of links with some constant, say 10: https://i.imgur.com/Yw4cjcB.png